Microsoft.DotNet.Interactive/PackageRestoreContext.cs

```diff
@@ -13,58 +13,71 @@
         private readonly HashSet<string> _restoreSources = new HashSet<string>();
-        private readonly Lazy<DirectoryInfo> _lazyDirectory;
+        private readonly Lazy<DependencyProvider> _lazyDependencies;
+        private readonly Lazy<IDependencyManagerProvider> _lazyIdm;
 
-        public PackageRestoreContext()
+        public PackageRestoreContext(string scriptExtension)
         {
-            _lazyDirectory = new Lazy<DirectoryInfo>(() =>
+            _scriptExtension = scriptExtension;
+            _lazyDependencies = new Lazy<DependencyProvider>(new DependencyProvider(AssemblyProbingPaths, NativeProbingRoots));
+            _lazyIdm = new Lazy<IDependencyManagerProvider>(_lazyDependencies.Value.TryFindDependencyManagerByKey(Enumerable.Empty<string>(), "", ReportError, packageKey));
+        }
+
+        private IEnumerable<string> AssemblyProbingPaths()
+        {
+            foreach (var package in _resolvedPackageReferences.Values)
             {
-                var dir = new DirectoryInfo(
-                    Path.Combine(
-                        Paths.UserProfile,
-                        ".net-interactive-csharp",
-                        Guid.NewGuid().ToString("N")));
+                foreach (var fi in package.AssemblyPaths)
+                    yield return fi.FullName;
+            }
+        }
 
-                if (!dir.Exists)
+        private IEnumerable<string> NativeProbingRoots ()
+        {
+            foreach (var package in _resolvedPackageReferences.Values)
+            {
+                foreach (var di in package.ProbingPaths)
                 {
-                    dir.Create();
+                    yield return di.FullName;
                 }
-
-                return dir;
-            });
-
-            AssemblyLoadContext.Default.Resolving += OnResolving;
+            }
         }
 
-        private Assembly OnResolving(AssemblyLoadContext loadContext, AssemblyName assemblyName)
+        private ResolvingErrorReport ReportError = (ErrorReportType errorType, int code, string message) =>
         {
-            var data = _resolvedPackageReferences.Values
-                .SelectMany(r => r.AssemblyPaths)
-                .Select(p => ( assemblyName: AssemblyName.GetAssemblyName(p.FullName), fileInfo:p )).ToList();
-            var found = data
-                .FirstOrDefault(a => a.assemblyName.FullName == assemblyName.FullName);
-
-            return found == default ? null : loadContext.LoadFromAssemblyPath(found.fileInfo.FullName);
-        }
-
-        public DirectoryInfo Directory => _lazyDirectory.Value;
+            if (errorType == ErrorReportType.Error)
+            {
+                Console.WriteLine("PackageManagementError {0} {1}", code, message);
+            }
+            else
+            {
+                Console.WriteLine("PackageManagementWarning {0} {1}", code, message);
+            }
+        };
```

Comment:
We should avoid writing directly the console here.

Reply:
This is a callback, that is invoked within the package resolution.  And so the console output goes to the "dotnet build process"

That console output is then retrieved and returned as part of the PackageRestoreResult

Here:
https://github.com/dotnet/interactive/pull/239/files#diff-c32393aaabb96f08f116d8d74d5edb32R220

I will add a comment to clarify that.